src/service-worker.ts

```diff
@@ -0,0 +1,79 @@
+/// <reference lib="webworker" />
+
+// Courtesy of https://dev.to/100lvlmaster/create-a-pwa-with-sveltekit-svelte-a36
+import { build, files, version } from '$service-worker';
+
+const worker = self as unknown as ServiceWorkerGlobalScope;
+const cacheName = `cache-${version}`;
+
+// `build` is an array of all the files generated by the bundler,
+// `files` is an array of everything in the `static` directory
+const toCache = build.concat(files);
+const staticAssets = new Set(toCache);
+
+worker.addEventListener('install', (event) => {
+  event.waitUntil(
+    caches
+      .open(cacheName)
+      .then((cache) => cache.addAll(toCache))
+      .then(() => worker.skipWaiting())
+  );
+});
+
+worker.addEventListener('activate', (event) => {
+  event.waitUntil(
+    caches.keys().then(async (keys) => {
+      // delete old caches
+      for (const key of keys) {
+        if (key !== cacheName) await caches.delete(key);
+      }
+
+      worker.clients.claim();
```

Comment:
issue (non-blocking?): I'm not sure but I think this code will not wait until the previous service workers are deactivated to activate the current one. This might be a problem, or it might not be. Not sure

Reply:
`claim` does not _activate_ the service worker. It is already activated, given that this code is inside the `activate` event handler. It is used exactly like that in the [`clients.claim` example on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Clients/claim#example).